File: w07_d2_scrabble/scrabble.rb
# Sam WS' code from Slack

require 'pry'

class Scrabble

  def initialize(word)
    @word = word
  end

  def self.score(word)
    letters = word.downcase.split("")

    total = 0

    letters.each do |letter|
      total += letter_score[letter]
    end

    return total
  end

  def self.letter_score
    {  "a"=>1, "b"=>3, "c"=>3, "d"=>2,
      "e"=>1, "f"=>4, "g"=>2, "h"=>4,
      "i"=>1, "j"=>8, "k"=>5, "l"=>1,
      "m"=>3, "n"=>1, "o"=>1, "p"=>3,
      "q"=>10, "r"=>1, "s"=>1, "t"=>1,
      "u"=>1, "v"=>4, "w"=>4, "x"=>8,
      "y"=>4, "z"=>10
    }
  end
end

binding.pry

File: w07_d2_scrabble/scrabble.js
// Write a program that, given a word, computes the scrabble score for that word.

// 1. split word into individual characters
// 2. run each character against the value table
// 3. add to final score
// 4. output final score

var scrabbleTable = {
  "A":1, "E":1, "I":1, "O":1, "U":1, "L":1, "N":1, "R":1, "S":1, "T":1,
  "D":2, "G":2,
  "B":3, "C":3, "M":3, "P":3,
  "F":4, "H":4, "V":4, "W":4, "Y":4,
  "K":5,
  "J":8, "X":8,
  "Q":10, "Z":10
};

console.log('Scrabble Score');

function ScrabbleScore(word) {
  var cleaned_word = word.replace(/[^\w\s]|_/g, "").replace(/\s+/g, "").toUpperCase();
  var length = cleaned_word.length;
  var split = cleaned_word.split('');
  var score = 0;

  for (var i = 0; i < length; i++) {
    score += (scrabbleTable[split[i]]);
  }
  return score;
}
